app/src/main/java/org/thoughtcrime/securesms/recoverypassword/RecoveryPassword.kt

```diff
@@ -176,10 +184,44 @@ private fun HideRecoveryPasswordCell(onHide: () -> Unit = {}) {
                     .align(Alignment.CenterVertically)
                     .contentDescription(R.string.AccessibilityId_recoveryPasswordHideRecoveryPassword),
                 color = LocalColors.current.danger,
-                onClick = onHide
+                onClick = { showHideRecoveryDialog = true }
             )
         }
     }
+
+    // recovery hide dialog
+    if(showHideRecoveryDialog) {
+        AlertDialog(
+            onDismissRequest = { showHideRecoveryDialog = false },
+            title = stringResource(R.string.recoveryPasswordHidePermanently),
+            text = stringResource(R.string.recoveryPasswordHidePermanentlyDescription1),
+            buttons = listOf(
+                DialogButtonModel(
+                    GetString(R.string.theContinue),
+                    color = LocalColors.current.danger,
+                    onClick = { showHideRecoveryConfirmationDialog = true }
+                ),
+                DialogButtonModel(GetString(android.R.string.cancel))
+            )
+        )
+    }
+
+    // recovery hide confirmation dialog
+    if(showHideRecoveryConfirmationDialog) {
+        AlertDialog(
+            onDismissRequest = { showHideRecoveryConfirmationDialog = false },
+            title = stringResource(R.string.recoveryPasswordHidePermanently),
+            text = stringResource(R.string.recoveryPasswordHidePermanentlyDescription2),
+            buttons = listOf(
+                DialogButtonModel(
+                    GetString(R.string.yes),
```

Comment:
Add button text as `qaTag` if req'd?

Reply:
The qaTag is applied automatically in the AlertDialog base class implementation. It is set to the text of the button there, on all buttons. Is that what you meant?